src/home-button/home-button.test.tsx

```diff
@@ -29,4 +46,37 @@ describe('home button', () => {
     userEvent.click(screen.getByLabelText('Home'))
     expect(window.location.href).toEqual('/bahmni/home')
   })
+
+  it('should update audit logs when user enters lab lite', async () => {
+    mockedOpenmrsFetch.mockReturnValueOnce({data: true})
+
+    render(
+      <SWRConfig value={{provider: () => new Map()}}>
+        <HomeButton />
+      </SWRConfig>,
+    )
+
+    await waitFor(() => expect(mockedOpenmrsFetch).toBeCalledTimes(2))
+
+    const auditMessagePayload = getPayloadForUserLogin(mockUser.username)
+    expect(mockedOpenmrsFetch.mock.calls[0][1].method).toBe('GET')
+    expect(mockedOpenmrsFetch.mock.calls[1][1].method).toBe('POST')
+    expect(mockedOpenmrsFetch.mock.calls[1][1].body).toBe(
+      JSON.stringify(auditMessagePayload),
+    )
+  })
```

Comment:
the mock.calls verification is way too ugly. Also you are not asserting on the API URL. In order to read better you should extract this into a function

```
  function verifyApiCall(
    url: string,
    requestType: string,
    body: string | undefined = undefined,
  ) {
    const callStatus = mockedOpenmrsFetch.mock.calls
      .filter(call => call[0] === url && call[1].method === requestType)
      .reduce((acc, call) => {
        if (call[1].method === 'GET') return acc || true
        if (body && call[1].body === body) return acc || true
        return acc
      }, false)

    if (!callStatus) console.log(`URL: ${url}, METHOD: ${requestType}, FAILED`)
    expect(callStatus).toBe(true)
  }
```

and then your assertions would look like
```
    verifyApiCall(
      '/ws/rest/v1/bahmnicore/sql/globalproperty?property=bahmni.enableAuditLog',
      'GET',
    )

    verifyApiCall(
      '/ws/rest/v1/auditlog',
      'POST',
      JSON.stringify(auditMessagePayload),
    )
```

Reply:
Changes have been made